python/cucim/src/cucim/skimage/transform/_geometric.py

```diff
@@ -1050,39 +1051,38 @@ def estimate(self, src, dst):
         Returns
         -------
         success : bool
-            True, if model estimation succeeds.
+            True, if all pieces of the model are successfully estimated.
 
         """
 
         ndim = src.shape[1]
         # forward piecewise affine
         # triangulate input positions into mesh
         xp = cp.get_array_module(src)
-        if xp is cp:
-            # TODO: grlee77 :update if spatial.Delaunay is implemented for GPU
-            # transfer to CPU for use of spatial.Delaunay
-            src = cp.asnumpy(src)
-            dst = cp.asnumpy(dst)
+        # TODO: update if spatial.Delaunay become available in CuPy
+        self._tesselation = spatial.Delaunay(cp.asnumpy(src))
```

Comment:
Should we open an issue about this?

Reply:
We could, it is probably not very easy to address though.  We would need a Delaunay triangulation class with a `vertices` attribute and `find_simplex` method for the GPU (There are several other attributes/methods on this class in SciPy, but they are not used here). On the SciPy side the implementation relies on the QHull C library.

Reply:
If we did find one, ideally it would be available via upstream `cupyx.scipy.spatial`

Reply:
Idk maybe it could happen. There has been work in the past like [CudaHull]( https://doi.org/10.1016/j.cag.2012.02.012 )